ksuid.go

```diff
@@ -206,7 +205,12 @@ func NewRandom() (ksuid KSUID, err error) {
 	copy(ksuid[timestampLengthInBytes:], randBuffer[:])
 
 	randMutex.Unlock()
+	return
+}
 
+// Generates a new KSUID
+func NewRandom() (ksuid KSUID, err error) {
```

Comment:
Should `NewRandom` be implemented as `NewRandomWithTime(time.Now())`? Then we don't need the `setRandBuffer` function.

Let me know if this is something you've already considered.